app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,5 @@
 class PostsController < ApplicationController
-  def show
-    @post = Post.new(current_user.token, params[:title], params[:post_id])
+  def index
+    @post = Post.build(current_user.token, params[:subreddit_name], params[:post_id])
```

Comment:
Build allows me to make many request calls and pull in all the data I need in a specific view. In that build method, I can create an instance of that model that turns the data hash into instance variables.